src/cdk/table/table.ts

```diff
@@ -556,21 +593,18 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
    * If there is a diff, then re-render that section.
    */
   private _renderUpdatedColumns() {
-    // Re-render the rows when the row definition columns change.
-    this._rowDefs.forEach(def => {
-      if (def.getColumnsDiff()) {
-        this._forceRenderRows();
-      }
-    });
+    const defColumnsDiffReducer = (accumulator, def) => accumulator || !!def.getColumnsDiff();
 
-    // Re-render the header row if there is a difference in its columns.
-    if (this._headerRowDef && this._headerRowDef.getColumnsDiff()) {
-      this._renderHeaderRow();
+    if (this._rowDefs.reduce(defColumnsDiffReducer, false)) {
+      this._forceRenderRows();
```

Comment:
Got out of sync in previous commits when renaming (current has a `renderRows` and `forceRenderRows`). Renamed to force for both of these since it clears the outlet and re-renders.